src/common/tracer.cc

```diff
@@ -10,8 +10,9 @@
 
 namespace tracing {
 
-const opentelemetry::nostd::shared_ptr<opentelemetry::trace::NoopTracer> Tracer::noop_tracer = opentelemetry::nostd::shared_ptr<opentelemetry::trace::NoopTracer>(new opentelemetry::trace::NoopTracer);
-const jspan Tracer::noop_span = opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span>(new opentelemetry::v1::trace::NoopSpan(Tracer::noop_tracer->shared_from_this()));
+const opentelemetry::nostd::shared_ptr<opentelemetry::trace::Tracer> Tracer::noop_tracer = opentelemetry::trace::Provider::GetTracerProvider()->GetTracer("no-op", OPENTELEMETRY_SDK_VERSION);
```

Comment:
why did you create a regular tracer here?

Reply:
that's how we create the noop tracer using opentelemetry
https://github.com/open-telemetry/opentelemetry-cpp/blob/main/api/include/opentelemetry/trace/provider.h#L28-L48

what I did first, was to avoid those calls and just create the shared_ptr by myself.

Reply:
just to make it clear,
the `NoopTracerProvider->GetTracer()` returns a static noop tracer. and then we use it the "start" a noop span

Reply:
is this happening only once or in the fast path?
if happening only once, we can use the official way of getting the tracer.

Reply:
@zenomri IIUC, this is not redefined anywhere else, would it have any consideration to use official way?

Reply:
@ideepika  
this is the official way. 
https://opentelemetry-cpp.readthedocs.io/en/latest/api/GettingStarted.html#obtain-a-tracer